n3fit/src/evolven3fit/evolven3fit_API.py

```diff
@@ -0,0 +1,209 @@
+    # read the runcard
+    my_runcard = utils.read_runcard(usr_path)
+    # theory_card construction
+    theory = Loader().check_theoryID(my_runcard["theory"]["theoryid"]).get_description()
+    theory.pop("FNS")
+    theory.update(t_card_dict)
+    t_card = gen_theory.gen_theory_card(theory["PTO"], theory["Q0"], update=theory)
+    # construct operator card
+    op_x_grid = utils.generate_x_grid()
+    q2_grid = utils.generate_q2grid(theory["Q0"], 1.0e5)
+    op_card = gen_op.gen_op_card(q2_grid, update={"interpolation_xgrid": op_x_grid})
+    op_card.update(op_card_dict)
+    return t_card, op_card
+
+# Temporary solution. Then it will be loaded from the theory itself
+def construct_eko_for_fit(t_card, op_card, save_path=None):
+    """
+    Construct the eko operator needed for evolution of fitted pdfs
+
+    Parameters
+    ----------
+        usr_path: pathlib.Path
+            path to the folder containing the fit  
+        save_path: pathlib.Path
+            path where the eko will be saved (the eko
+            won't be saved if save_path is None)
+    Returns
+    -------
+        : eko.output.Output
+        eko operator
+        : dict
+        theory runcard
+        : dict
+        operator runcard
+    """
+    
+    # generate eko operator (temporary because it will be loaded from theory)
+    eko_op = run_dglap(t_card, op_card)
+    if save_path is not None:
+        try:
+            eko_op.dump_tar(save_path)
+        except:
+            pass
+    return eko_op
+
+
+def evolve_exportgrid(exportgrid, eko, theory_card, operator_card):
+    """
+    Evolves the provided exportgrid for the desired replica with the eko and returns the evolved block
+
+    Parameters
+    ----------
+        exportgrid: dict
+            exportgrid of pdf at fitting scale
+        eko: eko object
+            eko operator for evolution
+        theory_card: dict
+            theory card
+        operator_card: dict
+            operator card
+    Returns
+    -------
+        : np.array
+        evolved block
+    """
+    # construct LhapdfLike object
+    pdf_grid = np.array(exportgrid["pdfgrid"]).transpose()
+    x_grid = np.array(exportgrid["xgrid"]).astype(np.float)
+    pdf_to_evolve = utils.LhapdfLike(pdf_grid, exportgrid["q20"], x_grid)
+    # evolve pdf
+    evolved_pdf = apply.apply_pdf(eko, pdf_to_evolve)
+    # generate block to dump
+    targetgrid = operator_card["interpolation_xgrid"]
+    block = genpdf.generate_block(
+        lambda pid, x, Q2, evolved_PDF=evolved_pdf: x * evolved_PDF[Q2]["pdfs"][pid][targetgrid.index(x)],
+        xgrid=targetgrid,
+        Q2grid=operator_card["Q2grid"],
+        pids=br.flavor_basis_pids,
+    )
+    return block
+
+
+def dump_evolved_replica(evolved_block, usr_path, replica_num):
+    path_where_dump = usr_path / "nnfit" / usr_path.stem
+    # create folder to dump the evolved replica if it does not exist
+    if not os.path.exists(path_where_dump):
+        os.makedirs(path_where_dump)
```

Comment:
You can use [this](https://docs.python.org/3/library/pathlib.html#pathlib.Path.mkdir) with `exists_ok=True`.